Server/ConcordiaCurriculumManagerTest/UnitTests/Services/UserAuthenticationServiceTests.cs

```diff
@@ -77,6 +78,84 @@ public async Task SigninUser_ValidUser_ReturnsAccessToken()
         Assert.IsNotNull(accessToken);
     }
 
+    [TestMethod]
+    public async Task SigninUser_ValidUserWithGroupMemberAndGroupMaster_ReturnsAccessTokenWithGroupAndGroupMasterClaims()
+    {
+        var userEmail = "test@example.com";
+        var userPassword = "password";
+        var hashedPassword = "hashedpassword";
+        var group1 = new Group() { Id = Guid.NewGuid(), Name = "TestGroup1" };
+        var group2 = new Group() { Id = Guid.NewGuid(), Name = "TestGroup2" };
+
+        var user = new User
+        {
+            FirstName = "fname",
+            LastName = "lname",
+            Email = userEmail,
+            Password = hashedPassword,
+            Groups = new List<Group>() { group1, group2 },
+            MasteredGroups = new List<Group> { group1 }
+        };
+
+        userRepository.Setup(repo => repo.GetUserByEmail(userEmail))
+            .ReturnsAsync(user);
+
+        inputHasher.Setup(hasher => hasher.Verify(userPassword, hashedPassword))
+            .Returns(true);
+
+        var accessToken = await userService.SigninUser(userEmail, userPassword);
+        var handler = new JwtSecurityTokenHandler();
+        var jwtSecurityToken = handler.ReadJwtToken(accessToken);
+        var groupClaims = jwtSecurityToken.Claims.Where(c => c.Type == Claims.Group).ToList();
+        var groupMasterClaims = jwtSecurityToken.Claims.Where(c => c.Type == Claims.GroupMaster).ToList();
+
+        Assert.IsNotNull(accessToken);
+        Assert.IsNotNull(groupClaims);
+        Assert.IsNotNull(groupMasterClaims);
+
+        for (int i = 0; i < user.Groups.Count; i++)
+        {
+            Assert.AreEqual(user.Groups[i].Id.ToString(), groupClaims[i].Value);
+        }
+
+        for (int i = 0; i < user.MasteredGroups.Count; i++)
```

Comment:
Are mastered groups shown as an array and not as a string even if there is only 1 group in the list?

Reply:
They are shown a string in the actual JWT token. The way I am decoding the token, every claim is an object so groupMasterClaims is a list of (a single) claim(s).